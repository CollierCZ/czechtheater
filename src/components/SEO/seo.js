import React from "react"
import PropTypes from "prop-types"
import Helmet from "react-helmet"
import { StaticQuery, graphql } from "gatsby"

export const PureSEO = ({ description,image,data,keywords,pageTitle }) => {
  const info = data.kontentItemBasicInfo.elements;
  const homeTitle = info.name.value;
  const favicon = info.favicon.value[0].url;
  const ogImage = image ? image : favicon;
  const metaDescription = description ? description : info.short_description.value;
  const title = pageTitle ? pageTitle : homeTitle;
  return (
    <Helmet
      htmlAttributes={{lang: "en"}}
      title={title}
      titleTemplate={`%s | ${homeTitle}`}
      link={[
        {"rel": "icon", 
        "type": "image/png", 
        "href": favicon
        }
      ]}
      meta={[
        {
          name: `description`,
          content: metaDescription,
        },
        {
          property: `og:title`,
          content: title,
        },
        {
          property: `og:description`,
          content: metaDescription,
        },
        {
          property: `og:image`,
          content: ogImage,
        },
        {
          property: `og:type`,
          content: `website`,
        },
        {
          name: `twitter:card`,
          content: `summary`,
        },
        {
          name: `twitter:creator`,
          content: `Czech Theater`,
        },
        {
          name: `twitter:title`,
          content: title,
        },
        {
          name: `twitter:description`,
          content: metaDescription,
        },
      ]
      .concat(
        keywords.length > 0
          ? {
              name: `keywords`,
              content: keywords.join(`, `),
            }
          : []
      )}
    />
  )
}

PureSEO.defaultProps = {
  keywords: [`theater`, `czech`, `amateur`],
  pageTitle: "Czech Theater"
}

PureSEO.propTypes = {
  data: PropTypes.object.isRequired,
  description: PropTypes.string,
  image: PropTypes.string,
  keywords: PropTypes.arrayOf(PropTypes.string),
  pageTitle: PropTypes.string.isRequired,
}

function SEO({ description, image, keywords, title }) {
  return (
    <StaticQuery
      query={graphql`
        query SEOQuery {
          kontentItemBasicInfo {
            ...BasicInfoFragment
          }
        }
      `}
      render={data => {
        return (
          <PureSEO description={description} image={image} data={data} keywords={keywords} pageTitle={title} /> 
        )
      }}
    />
  )
}

SEO.propTypes = {
  description: PropTypes.string,
  image: PropTypes.string,
  keywords: PropTypes.arrayOf(PropTypes.string),
  title: PropTypes.string,
}

export default SEO