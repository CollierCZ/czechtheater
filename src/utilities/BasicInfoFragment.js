import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KenticoCloudItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedHtml
            }
            favicon {
                assets{
                  url
                }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;


