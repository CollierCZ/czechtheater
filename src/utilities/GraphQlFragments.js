import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KontentItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedData {
                html
              }
            }
            favicon {
              value {
                url
              }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;

export const ShowListFragment = graphql`
    fragment ShowListFragment on KontentItemShowSection {
      elements {
        shows {
          linked_items {
            elements {
              main_image {
                value {
                  url
                  description
                }
              }
              name {
                value
              }
              url {
                value
              }
              short_description {
                value
              }
              ticket_link {
                value
              }
              gallery {
                value {
                  url
                  description
                }
              }
            }
          }
        }
      }
    }
`;


