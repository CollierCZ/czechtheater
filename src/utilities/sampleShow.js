export const sampleShow = {
    elements: {
      name: {
        value: "Alquist's revenge"
      },
      description: {
        resolvedHtml: "All the acting"
      },
      main_image: {
        assets: [{
          description: "A cool image",
          url: "https://example.com"
        }]
      },
      ticket_link: {
        value: "https://example.com"
      },
      url: {
        value: "alquists-revenge"
      },
      gallery: {
          assets: [
              {
                  description: "A cool image",
                  url: "https://example.com"
              }
          ]
      }
    }
  }