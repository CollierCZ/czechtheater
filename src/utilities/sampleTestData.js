export const sampleShow = {
  elements: {
    name: {
      value: "Alquist's revenge"
    },
    description: {
      resolvedHtml: "<p>All the acting</p>"
    },
    main_image: {
      assets: [{
        description: "A cool image",
        url: "https://example.com"
      }]
    },
    ticket_link: {
      value: "https://example.com"
    },
    url: {
      value: "alquists-revenge"
    },
    gallery: {
      assets: [{
        description: "A cool image",
        url: "https://example.com"
      }]
    }
  }
}

export const sampleShowNoGallery = JSON.parse(JSON.stringify(sampleShow))
sampleShowNoGallery.gallery.assets = []

export const sampleShowNoTicketLink = JSON.parse(JSON.stringify(sampleShow))
sampleShowNoGallery.ticket_link.value = ""

export const basicInfoQuery = {
  kenticoCloudItemBasicInfo: {
    elements: {
      name: {
        value: "Czech Theater"
      },
      favicon: {
        assets: [{
          url: "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg"
        }]
      },
      short_description: {
        value: "A great theater."
      }
    }
  }
}