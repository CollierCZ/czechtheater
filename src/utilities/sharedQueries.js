import { graphql } from "gatsby";

export const basicInfoQuery = graphql`
  query BasicInfoQuery {
    kenticoCloudItemBasicInfo {
      elements {
        name {
          value
        }
        short_description {
          value
        }
        favicon {
          assets{
            url
          }
        }
      }
    }
  }
`