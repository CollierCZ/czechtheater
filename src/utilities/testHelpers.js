import { shallow } from "enzyme"

export const simpleShallowRender = (component) => {
  const wrapper = shallow (component)
  expect(wrapper).toMatchSnapshot()
}