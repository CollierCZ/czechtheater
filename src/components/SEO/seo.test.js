import { basicInfoQuery } from "../../utilities/sampleTestData" 
import React from "react"
import { shallow } from "enzyme"
import { PureSEO as SEO } from "./seo"

describe("SEO", () => {
  it("renders correctly with no additional props", () => {
    const component = shallow (<SEO data={basicInfoQuery} />)
    expect(component).toMatchSnapshot()
  })
  it("renders correctly with description", () => {
    const component = shallow (<SEO  data={basicInfoQuery} description="An awesome play" />)
    expect(component).toMatchSnapshot()
  })
  it("renders correctly with image", () => {
    const component = shallow (<SEO  data={basicInfoQuery} image="https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/2ad1a0d2-3809-4eb6-9024-d782425209f0/Poster.jpg" />)
    expect(component).toMatchSnapshot()
  })
  it("renders correctly with title", () => {
    const component = shallow (<SEO  data={basicInfoQuery} title="Alquist's Revenge" />)
    expect(component).toMatchSnapshot()
  })
})