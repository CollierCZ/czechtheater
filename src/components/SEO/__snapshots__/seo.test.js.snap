// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO renders correctly with description 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "en",
    }
  }
  link={
    Array [
      Object {
        "href": "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg",
        "rel": "icon",
        "type": "image/png",
      },
    ]
  }
  meta={
    Array [
      Object {
        "content": "An awesome play",
        "name": "description",
      },
      Object {
        "content": "Czech Theater",
        "property": "og:title",
      },
      Object {
        "content": "An awesome play",
        "property": "og:description",
      },
      Object {
        "content": "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg",
        "property": "og:image",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
      Object {
        "content": "Czech Theater",
        "name": "twitter:creator",
      },
      Object {
        "content": "Czech Theater",
        "name": "twitter:title",
      },
      Object {
        "content": "An awesome play",
        "name": "twitter:description",
      },
      Object {
        "content": "theater, czech, amateur",
        "name": "keywords",
      },
    ]
  }
  title="Czech Theater"
  titleTemplate="%s | Czech Theater"
/>
`;

exports[`SEO renders correctly with image 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "en",
    }
  }
  link={
    Array [
      Object {
        "href": "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg",
        "rel": "icon",
        "type": "image/png",
      },
    ]
  }
  meta={
    Array [
      Object {
        "content": "A great theater.",
        "name": "description",
      },
      Object {
        "content": "Czech Theater",
        "property": "og:title",
      },
      Object {
        "content": "A great theater.",
        "property": "og:description",
      },
      Object {
        "content": "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/2ad1a0d2-3809-4eb6-9024-d782425209f0/Poster.jpg",
        "property": "og:image",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
      Object {
        "content": "Czech Theater",
        "name": "twitter:creator",
      },
      Object {
        "content": "Czech Theater",
        "name": "twitter:title",
      },
      Object {
        "content": "A great theater.",
        "name": "twitter:description",
      },
      Object {
        "content": "theater, czech, amateur",
        "name": "keywords",
      },
    ]
  }
  title="Czech Theater"
  titleTemplate="%s | Czech Theater"
/>
`;

exports[`SEO renders correctly with no additional props 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "en",
    }
  }
  link={
    Array [
      Object {
        "href": "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg",
        "rel": "icon",
        "type": "image/png",
      },
    ]
  }
  meta={
    Array [
      Object {
        "content": "A great theater.",
        "name": "description",
      },
      Object {
        "content": "Czech Theater",
        "property": "og:title",
      },
      Object {
        "content": "A great theater.",
        "property": "og:description",
      },
      Object {
        "content": "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg",
        "property": "og:image",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
      Object {
        "content": "Czech Theater",
        "name": "twitter:creator",
      },
      Object {
        "content": "Czech Theater",
        "name": "twitter:title",
      },
      Object {
        "content": "A great theater.",
        "name": "twitter:description",
      },
      Object {
        "content": "theater, czech, amateur",
        "name": "keywords",
      },
    ]
  }
  title="Czech Theater"
  titleTemplate="%s | Czech Theater"
/>
`;

exports[`SEO renders correctly with title 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "en",
    }
  }
  link={
    Array [
      Object {
        "href": "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg",
        "rel": "icon",
        "type": "image/png",
      },
    ]
  }
  meta={
    Array [
      Object {
        "content": "A great theater.",
        "name": "description",
      },
      Object {
        "content": "Czech Theater",
        "property": "og:title",
      },
      Object {
        "content": "A great theater.",
        "property": "og:description",
      },
      Object {
        "content": "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg",
        "property": "og:image",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
      Object {
        "content": "Czech Theater",
        "name": "twitter:creator",
      },
      Object {
        "content": "Czech Theater",
        "name": "twitter:title",
      },
      Object {
        "content": "A great theater.",
        "name": "twitter:description",
      },
      Object {
        "content": "theater, czech, amateur",
        "name": "keywords",
      },
    ]
  }
  title="Czech Theater"
  titleTemplate="%s | Czech Theater"
/>
`;
